src/awkward/operations/ak_from_buffers.py

```diff
@@ -73,10 +73,12 @@ def from_buffers(
             behavior=behavior,
         ),
     ):
-        return _impl(form, length, container, buffer_key, backend, highlevel, behavior)
+        return _impl(
+            form, length, container, buffer_key, backend, highlevel, behavior, False
```

Comment:
Shouldn't this be `True` for the user-facing function?
```suggestion
            form, length, container, buffer_key, backend, highlevel, behavior, True
```

Reply:
No, it's important that the user-facing function does not change the user's Forms. Users would be very surprised if they passed a Form to `ak.from_buffers` and they got an array with a different Form.

Originally, this back door was called `from_ArrayBuilder`, but non-canonical Forms can also come from AwkwardForth.

The name could even be `backdoor`, but we don't have to decide that now.

Reply:
That's a good point. So, what we're doing here is allowing someone to user `from_buffers` if they have a non-canonical form, but we ensure that the result _is_ canonical. 

This is a bad default for users, who likely don't want the form to be coerced. I think we should expose this to the users, then - it never results in non-canonical layouts. It just let's them use non-canonical representations, which I think is reasonable; it's perfectly sound to express such a tree, we just don't support any operations on it.

Maybe we can call this parameter `strict=True` at the high-level, so users can say "I don't care if the array type changes" as an opt-in?